src/tracks/playabletrack/wavetrack/ui/WaveClipAdjustBorderHandle.cpp

```diff
@@ -75,19 +75,59 @@ namespace {
     virtual void Cancel() = 0;
 
     virtual void Draw(
-        TrackPanelDrawingContext &context, 
-        const wxRect &rect, 
+        TrackPanelDrawingContext &context,
+        const wxRect &rect,
         unsigned iPass);
 
     virtual wxRect DrawingArea(
-        TrackPanelDrawingContext&, 
-        const wxRect &rect, 
-        const wxRect &panelRect, 
+        TrackPanelDrawingContext&,
+        const wxRect &rect,
+        const wxRect &panelRect,
         unsigned iPass);
  };
 
 WaveClipAdjustBorderHandle::AdjustPolicy::~AdjustPolicy() = default;
 
+namespace {
+double GetLeftAdjustLimit(
+   const WaveClip& clip, const WaveTrack& track, bool adjustingLeftBorder,
+   bool isStretchMode)
+{
+   if (!adjustingLeftBorder)
+      // A clip of one sample is invisible - who needs that ? Leave at least 2.
+      return clip.GetPlayStartTime() + 2.0 / clip.GetRate();
+
+   const auto prevClip = track.GetNextClip(clip, PlaybackDirection::backward);
+   if (isStretchMode)
+      return prevClip ? prevClip->GetPlayEndTime() :
+                        std::numeric_limits<double>::min();
```

Comment:
Beware the difference of `min` and `lowest` for floating-point types!

Reply:
Great catch, thank you!